akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -186,6 +186,9 @@ import scala.util.control.NonFatal
             case Inert ⇒ // nothing to be done
             case _     ⇒ ErrorPublisher(ex, "failed-VirtualProcessor").subscribe(s)
           }
+        case _ if t == null ⇒
+          // cancelled before onError(null), must throw NPE, rule 2:13
+          throw ex
```

Comment:
Hmm, might be good to throw `ReactiveStreamsCompliance.exceptionMustNotBeNullException` instead so it's more explanatory than a raw NPE

Reply:
nevermind, just noticed the line below which github had folded, it's all good